### PR TITLE
Restructure logs.py into a module; include log_cycle

### PR DESCRIPTION
**Pull request by Pwuts**
Part of #4799

Title speaks for itself :)

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```

### CODE REVIEW

File: autogpt/agent/agent.py [hunk 1]
```diff
@@ -12,13 +12,13 @@
 from autogpt.llm.chat import chat_with_ai
 from autogpt.llm.providers.openai import OPEN_AI_CHAT_MODELS
 from autogpt.llm.utils import count_string_tokens
-from autogpt.log_cycle.log_cycle import (
+from autogpt.logs import logger, print_assistant_thoughts, remove_ansi_escape
```
Comment (collijk) [inline autogpt/agent/agent.py]: Everything is in the logs namespace now.
```suggestion
from autogpt.logs import (
    logger, 
    print_assistant_thoughts,
    remove_ansi_escape,
    FULL_MESSAGE_HISTORY_FILE_NAME,
    NEXT_ACTION_FILE_NAME,
    USER_INPUT_FILE_NAME,
    LogCycleHandler,
```

File: autogpt/llm/chat.py [hunk 2]
```diff
@@ -12,8 +12,8 @@
 from autogpt.llm.api_manager import ApiManager
 from autogpt.llm.base import ChatSequence, Message
 from autogpt.llm.utils import count_message_tokens, create_chat_completion
-from autogpt.log_cycle.log_cycle import CURRENT_CONTEXT_FILE_NAME
 from autogpt.logs import logger
```
Comment (collijk) [inline autogpt/llm/chat.py]:
```suggestion
from autogpt.logs import logger, CURRENT_CONTEXT_FILE_NAME
```

File: autogpt/memory/message_history.py [hunk 3]
```diff
@@ -13,8 +13,8 @@
 from autogpt.llm.base import ChatSequence, Message, MessageRole, MessageType
 from autogpt.llm.providers.openai import OPEN_AI_CHAT_MODELS
 from autogpt.llm.utils import count_string_tokens, create_chat_completion
-from autogpt.log_cycle.log_cycle import PROMPT_SUMMARY_FILE_NAME, SUMMARY_FILE_NAME
 from autogpt.logs import logger
```
Comment (collijk) [inline autogpt/memory/message_history.py]:
```suggestion
from autogpt.logs import logger, PROMPT_SUMMARY_FILE_NAME, SUMMARY_FILE_NAME
```

**Review by collijk — APPROVED** [review]
Some optional suggestions here.  I'm presuming the only changes here is code motion + typing (nice job on that!).  If you've added any logic changes, please flag them with a comment and re-request a review.  Otherwise looks good!

File: autogpt/logs/utils.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    if assistant_thoughts:
+        assistant_thoughts_reasoning = remove_ansi_escape(
+            assistant_thoughts.get("reasoning", "")
+        )
+        assistant_thoughts_plan = remove_ansi_escape(assistant_thoughts.get("plan", ""))
+        assistant_thoughts_criticism = remove_ansi_escape(
+            assistant_thoughts.get("criticism", "")
+        )
+        assistant_thoughts_speak = remove_ansi_escape(
+            assistant_thoughts.get("speak", "")
```
Comment (Pwuts) [inline autogpt/logs/utils.py:36]: Added default empty strings here to prevent errors in `remove_ansi_escape` when the LLM omits items in the response.

**Review by collijk — APPROVED** [review]

**Pull request merged**